src/oas2/__tests__/accessors.test.ts

```diff
@@ -20,13 +22,90 @@ const securityDefinitionsFixture: Dictionary<Security> = {
 };
 
 describe('accessors', () => {
-  const securityFixture = [
+  const securityFixture: Security[] = [
     {
       api_key: [],
       petstore_auth: ['write:pets', 'read:pets'],
     },
   ];
 
+  describe('relation between schemes', () => {
+    describe('when all fo the given schemes are expected to be validated against', () => {
+      const securityFixtureWithAndRelation = securityFixture;
+
+      it('returns an array containing multiple elements', () => {
+        expect(
+          getSecurities(
+            {
+              securityDefinitions: securityDefinitionsFixture,
+              security: [],
+            },
+            securityFixtureWithAndRelation,
+          ),
+        ).toEqual([
+          [
+            {
+              in: 'header',
+              name: 'api_key',
+              type: 'apiKey',
+            },
+            {
+              authorizationUrl: 'http://swagger.io/api/oauth/dialog',
+              flow: 'implicit',
+              scopes: {
+                'read:pets': 'read your pets',
+                'write:pets': 'modify pets in your account',
+              },
+              type: 'oauth2',
+            },
+          ],
+        ]);
+      });
+    });
+
+    describe('when one fo the given schemes is expected to be validated against', () => {
+      it('returns arrays containing one element each', () => {
+        const securityFixtureWithOrRelation: Security[] = [
+          {
+            petstore_auth: ['write:pets', 'read:pets'],
+          },
+          {
+            api_key: [],
+          },
```

Comment:
If I understand correctly this is not an OR relationship, this is *AND*

Reply:
@XVincentX, this is `OR` (https://github.com/stoplightio/http-spec/blob/40232d757aca32b5ceb0a35d152b04a5842111ff/src/oas2/__tests__/accessors.test.ts#L68). And this https://github.com/stoplightio/http-spec/blob/40232d757aca32b5ceb0a35d152b04a5842111ff/src/oas2/__tests__/accessors.test.ts#L34 is `AND`. Hope this helps!

Reply:
👍